app/uk/gov/hmrc/gform/auth/AuthService.scala

```diff
@@ -67,6 +72,35 @@ class AuthService(
         performAgent(agentAccess, formTemplate, lang, ggAuthorised(RecoverAuthResult.noop), ifSuccessPerformEnrolment)
     }
 
+  private def performAWSALBAuth()(implicit hc: HeaderCarrier): AuthResult = {
+    val encodedJWT: Seq[(String, String)] = hc.otherHeaders.filter(header => header._1 == "x-amzn-oidc-data")
+
+    encodedJWT.headOption match {
+      case None => AuthBlocked("You are not authorized to access this service")
+      case Some((_, jwt)) => {
+        val decoder = Base64.getDecoder
+        val splitJwt = jwt.split("\\.")
+        if (!(splitJwt.length == 3)) {
+          Logger.error(s"Corrupt JWT received from AWS ALB: [$jwt]")
+          AuthBlocked("You are not authorized to access this service")
+        } else {
+          val payload = new String(decoder.decode(splitJwt(1)))
+          Json.parse(payload) \ "username" match {
+            case JsDefined(value) =>
```

Comment:
You can use constructor pattern matching. It would looks like:
```
case JsDefined(JsString(username)) => AuthSuccessful(AWSALBRetrievals(username))
case JsDefined(_)                  => Logger.error(s"Corrupt JWT received from AWS ALB: [$jwt]")
                                      AuthBlocked("You are not authorized to access this service")
case JsUndefined()                 => AuthBlocked("Username does not exist in JWT")
```